### DISCUSSION TITLE
How do I completley reinstall winget when it is corrupted

### DISCUSSION THREAD

**Discussion by YeesterPlus** [microsoft/winget-cli#4801] [Q&A] [answered]
WINGET IS CORRUPTED, PLEASE HELP

**Comment by mdanish-kh** [comment]
Can you explain a bit more about the problem you're having? Are you able to run a command like `winget --help` or `winget list`? If not, can you share the error output?

**Reply by Daniel-Shimizu** [reply, marked as answer]
Hi, 
In my case I am getting the message when trying to upgrade
0x8a15003f : The source data is corrupted or tampered

I find a solution 
https://github.com/microsoft/winget-cli/issues/4219#issuecomment-1975125826


Thanks

**Comment by denelon** [comment]
You could try running the `Repair-WinGetPackageManager` cmdlet inside the [Microsoft.WinGet.Client](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client) PowerShell module to ensure that WinGet is correctly installed.

If that isn't the issue, then you can try to `winget source update --source winget`
